infrastructures/openshift/src/test/java/org/eclipse/che/workspace/infrastructure/openshift/OpenShiftInternalRuntimeTest.java

```diff
@@ -0,0 +1,394 @@
+import org.eclipse.che.api.workspace.shared.dto.event.MachineStatusEvent;
+import org.eclipse.che.dto.server.DtoFactory;
+import org.eclipse.che.workspace.infrastructure.openshift.bootstrapper.OpenShiftBootstrapper;
+import org.eclipse.che.workspace.infrastructure.openshift.bootstrapper.OpenShiftBootstrapperFactory;
+import org.eclipse.che.workspace.infrastructure.openshift.environment.OpenShiftEnvironment;
+import org.eclipse.che.workspace.infrastructure.openshift.project.OpenShiftPersistentVolumeClaims;
+import org.eclipse.che.workspace.infrastructure.openshift.project.OpenShiftPods;
+import org.eclipse.che.workspace.infrastructure.openshift.project.OpenShiftProject;
+import org.eclipse.che.workspace.infrastructure.openshift.project.OpenShiftRoutes;
+import org.eclipse.che.workspace.infrastructure.openshift.project.OpenShiftServices;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+/**
+ * Tests {@link OpenShiftInternalRuntime}.
+ *
+ * @author Anton Korneta
+ */
+public class OpenShiftInternalRuntimeTest {
+
+  private static final int EXPOSED_PORT_1 = 4401;
+  private static final int EXPOSED_PORT_2 = 8081;
+  private static final int NOT_EXPOSED_PORT_1 = 4411;
+
+  private static final String WORKSPACE_ID = "workspace123";
+  private static final String PVC_NAME = "che-workspace-data";
+  private static final String POD_NAME = "app";
+  private static final String ROUTE_NAME = "test-route";
+  private static final String SERVICE_NAME = "test-service";
+  private static final String POD_SELECTOR = "che.pod.name";
+  private static final String CONTAINER_NAME_1 = "test1";
+  private static final String CONTAINER_NAME_2 = "test2";
+  private static final String ROUTE_HOST = "localhost";
+  private static final String M1_NAME = POD_NAME + '/' + CONTAINER_NAME_1;
+  private static final String M2_NAME = POD_NAME + '/' + CONTAINER_NAME_2;
+
+  private static final RuntimeIdentity IDENTITY =
+      new RuntimeIdentityImpl(WORKSPACE_ID, "env1", "usr1");
+
+  @Mock private OpenShiftRuntimeContext context;
+  @Mock private EventService eventService;
+  @Mock private ServerCheckerFactory serverCheckerFactory;
+  @Mock private OpenShiftBootstrapperFactory bootstrapperFactory;
+  @Mock private OpenShiftEnvironment osEnv;
+  @Mock private OpenShiftProject project;
+  @Mock private OpenShiftPersistentVolumeClaims pvcs;
+  @Mock private OpenShiftServices services;
+  @Mock private OpenShiftRoutes routes;
+  @Mock private OpenShiftPods pods;
+  @Mock private OpenShiftBootstrapper bootstrapper;
+  @Mock private InternalEnvironment environment;
+
+  @Captor private ArgumentCaptor<MachineStatusEvent> eventCaptor;
+
+  private OpenShiftInternalRuntime internalRuntime;
+
+  @BeforeMethod
+  public void setup() throws Exception {
+    MockitoAnnotations.initMocks(this);
```

Comment:
Why do you init mocks manually instead of using MockitoTestNGListener?

Reply:
Because we have problems with `ArgumentCaptor` it's not empty after the test method execution

Reply:
I see. Thanks

Reply:
Should we create an issue in Mockito-TestNG connector project?

Reply:
Maybe but not in the scope of this issue.

Reply:
I'll create the issue for that.

Reply:
created issue https://github.com/skabashnyuk/mockitong/issues/1

Reply:
Wow, the first one! Thanks!